lib/runner.js

```diff
@@ -386,6 +386,15 @@ Runner.prototype.hook = function(name, fn) {
       }
       if (err) {
         if (err instanceof Pending) {
+          if (name === HOOK_TYPE_AFTER_EACH || name === HOOK_TYPE_AFTER_ALL) {
+            utils.deprecate(
+              util.format(
+                'skipping a test within an "%s" hook is DEPRECATED and will throw an exception in a future version of Mocha. ' +
+                  'Use a return statement or other means to abort hook execution.',
+                name === HOOK_TYPE_AFTER_EACH ? 'after each' : 'after all'
```

Comment:
Are these names (terms) being used to have a generic message for all interfaces?

* TDD uses `teardown` and `suiteTeardown`
* other provided ones use `after` and `afterEach`

Reply:
yes, meant as generic term for all interfaces.